Automattic/Gravatar-SDK-Android
# `gravatar-ui`- Refactor text styling

Closes #

### Description

In this PR, I'm removing the use of `ProvideTextStyle` in our Profile components by adding a `textStyle` param in the atomic components.

I think this makes it easier to customize the atomic components and also has a `Default` style for the atomic ones. 

Another point about `ProvideTextStyle` is that it changes the `LocalTextStyle` for all the `@Composables` below it. This issue is easy to see in our `ExpandableText`:

<img width="1165" alt="image" src="https://github.com/Automattic/Gravatar-SDK-Android/assets/6974554/1a3448d4-c5f8-4bdf-a284-787f75b74cb9">

The dialog text uses the style defined in the `ProvideTextStyle`.

Providing a `TextStyle` as a param with a default value (that should be a _GravatarStyle_) has fewer side effects and makes it more understandable to me. 

**Note:** This is a proposal to discuss; it may have some cons that I'm not seeing right now. 

### Testing Steps

- Run the DemoApp and verify everything in the Profile tab looks as expected.

| Before | After |
|----------|-----------|
|![old](https://github.com/Automattic/Gravatar-SDK-Android/assets/6974554/3fe54bb4-7b70-480f-91cb-44a16d93459c)|![new](https://github.com/Automattic/Gravatar-SDK-Android/assets/6974554/5a1cf2d0-c187-4443-9460-27df69bbda3c)|
